eng/Version.Details.xml

```diff
@@ -11,283 +11,283 @@
   <ProductDependencies>
     <Dependency Name="Microsoft.AspNetCore.Blazor.Mono" Version="3.2.0-preview1.20067.1">
       <Uri>https://github.com/aspnet/Blazor</Uri>
-      <Sha>dd7fb4d3931d556458f62642c2edfc59f6295bfb</Sha>
+      <Sha>7868699de745fd30a654c798a99dc541b77b95c0</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.Razor.Language" Version="3.1.0-rtm.19573.1">
-      <Uri>https://github.com/aspnet/AspNetCore-Tooling</Uri>
-      <Sha>abd3c17b317dbe455b18a4f5937c43e0fc7a2944</Sha>
+    <Dependency Name="Microsoft.AspNetCore.Razor.Language" Version="3.1.2">
+      <Uri>https://github.com/dotnet/aspnetcore-tooling</Uri>
+      <Sha>2dab42e151ea8020a75cdaaa8c46bf5d9093b8c0</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.Mvc.Razor.Extensions" Version="3.1.0-rtm.19573.1">
-      <Uri>https://github.com/aspnet/AspNetCore-Tooling</Uri>
-      <Sha>abd3c17b317dbe455b18a4f5937c43e0fc7a2944</Sha>
+    <Dependency Name="Microsoft.AspNetCore.Mvc.Razor.Extensions" Version="3.1.2">
+      <Uri>https://github.com/dotnet/aspnetcore-tooling</Uri>
+      <Sha>2dab42e151ea8020a75cdaaa8c46bf5d9093b8c0</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.CodeAnalysis.Razor" Version="3.1.0-rtm.19573.1">
-      <Uri>https://github.com/aspnet/AspNetCore-Tooling</Uri>
-      <Sha>abd3c17b317dbe455b18a4f5937c43e0fc7a2944</Sha>
+    <Dependency Name="Microsoft.CodeAnalysis.Razor" Version="3.1.2">
+      <Uri>https://github.com/dotnet/aspnetcore-tooling</Uri>
+      <Sha>2dab42e151ea8020a75cdaaa8c46bf5d9093b8c0</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NET.Sdk.Razor" Version="3.1.0-rtm.19573.1">
-      <Uri>https://github.com/aspnet/AspNetCore-Tooling</Uri>
-      <Sha>abd3c17b317dbe455b18a4f5937c43e0fc7a2944</Sha>
+    <Dependency Name="Microsoft.NET.Sdk.Razor" Version="3.1.2">
+      <Uri>https://github.com/dotnet/aspnetcore-tooling</Uri>
+      <Sha>2dab42e151ea8020a75cdaaa8c46bf5d9093b8c0</Sha>
     </Dependency>
-    <Dependency Name="dotnet-ef" Version="3.1.0" Pinned="true">
-      <Uri>https://github.com/aspnet/EntityFrameworkCore</Uri>
-      <Sha>82c6ea483d3a17393dad5986df9a8eef89ddcd07</Sha>
+    <Dependency Name="dotnet-ef" Version="3.1.2">
+      <Uri>https://github.com/dotnet/efcore</Uri>
+      <Sha>7ae89d6adb7433b6ae9c37091e57a1226092ef46</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.EntityFrameworkCore.InMemory" Version="3.1.0" Pinned="true">
-      <Uri>https://github.com/aspnet/EntityFrameworkCore</Uri>
-      <Sha>82c6ea483d3a17393dad5986df9a8eef89ddcd07</Sha>
+    <Dependency Name="Microsoft.EntityFrameworkCore.InMemory" Version="3.1.2">
```

Comment:
Huh? Why isn't almost-every dependency in this file pinned?

Also:
- doubt even 1/3 of these dependencies are needed
- make sure none of the dependencies use `CoherentParentDependency` and `Pinned` together

Reply:
I have no idea what Pinned means. I copied this file pretty-much as is from the release/3.1 branch.

Reply:
This looks fine now. Because we don't have subscriptions from the extensions or efcore repo, `Pinned` is irrelevant. That is, unless you need to pin something coming from blazor, you could remove all `Pinned` attributes.